chaosaws/ec2/actions.py

```diff
@@ -657,3 +657,106 @@ def attach_instance_volume(client: boto3.client,
             'Unable to attach volume %s to instance %s: %s' % (
                 volume_id, instance_id, e.response['Error']['Message']))
     return response
+
+
+def authorize_security_group_ingress(requested_security_group_id: str,
+                                     ip_protocol: str,
+                                     from_port: int,
+                                     to_port: int,
+                                     ingress_security_group_id: str = None,
+                                     cidr_ip: str = None,
+                                     configuration: Configuration = None,
+                                     secrets: Secrets = None) -> AWSResponse:
+    """
+    Add one ingress rule to a security group
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Client.authorize_security_group_ingress
+
+    - requested_security_group_id: the id for the security group to update
+    - ip_protocol: ip protocol name (tcp, udp, icmp, icmpv6) or -1 to specify all
+    - from_port: start of port range
+    - to_port: end of port range
+    - ingress_security_group_id: id of the securiy group to allow access to. You can either specify this or cidr_ip.
+    - cidr_ip: the IPv6 CIDR range.
+    You can either specify this or ingress_security_group_id
+    """  # noqa: E501
+    client = aws_client('ec2', configuration, secrets)
+    request_kwargs = {
+        'GroupId': requested_security_group_id,
+        'IpPermissions': [
+            {
+                'IpProtocol': ip_protocol,
+                'IpRanges': [{
+                    # conditionally assign the following
+                    # 'CidrIp': cidr_ip
+                }],
+                'FromPort': from_port,
+                'ToPort': to_port,
+                'UserIdGroupPairs': [{
+                    # conditionally assign the following
+                    # 'GroupId': ingress_security_group_id
+                }]
+            }
+        ]
+    }
+    if cidr_ip is not None:
+        request_kwargs['IpPermissions'][0]['IpRanges'][0]['CidrIp'] = cidr_ip  # noqa: E501
+    if ingress_security_group_id is not None:
+        request_kwargs['IpPermissions'][0]['UserIdGroupPairs'][0]['GroupId'] = ingress_security_group_id  # noqa: E501
+    try:
+        response = client.authorize_security_group_ingress(**request_kwargs)
+        return response
+    except ClientError as e:
+        raise FailedActivity(
```

Comment:
Please use `ActivityFailed` as it replaced `FailedActivity` (the latter is deprecated and will be removed someday).

Reply:
noted, done